### ISSUE TITLE
Highlight type restrictions

### ISSUE THREAD

**Issue by jpellerin** [jpellerin/emacs-crystal-mode#1]

**Comment by rocky** [comment]
Is this still an issue? When I look at the first example in http://crystal-lang.org/docs/syntax_and_semantics/type_restrictions.html  

``` cryrstal
def add(x : Number, y : Number)
  x + y
end
```

Number is in a different font.

**Comment by jpellerin** [comment]
It sort of works by accident, because of the capitalization I think. There's no special support for giving the restriction in a different face. Or at least I don't remember adding any! But if it is working well enough as-is then I'm ok with closing this issue.

**Comment by elthariel** [comment]
Nah, I had a small look and as far as i understand elisp, there's no support for it in the parsers.
That's also a problem when you do

``` ruby
alias NewType = Int8*
class Test
  def test
  end
end
```
